### PR TITLE
SAK-43319 Fixed various grade types in Grader.

### PR DESCRIPTION
**Pull request by adrianfish**
https://jira.sakaiproject.org/browse/SAK-43319

Reusing Assignments i18n bundle

dialog feedback working ok
inline feedback working
private notes and cancel working better
rubric saving working
Added rubric grader button
fixed date picker

### CODE REVIEW

**Comment by mpellicer** [comment]
@adrianfish what's the reason for moving the properties? The Spanish group is translating 20 right now for Catalonian, Basque and Spanish, this is quite problematic in our transifex sync process.

https://www.transifex.com/apereo/sakai-trunk/

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 7
           

Complexity increasing per file
==============================
- rubrics/tool/src/main/frontend/webcomponents/sakai-rubric-grading-button.js  5
         

Complexity decreasing per file
==============================
+ webcomponents/tool/src/main/frontend/grader/sakai-grader.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5124965&bid=16846229)

**Comment by adrianfish** [comment]
I've moved the props so we can pick them up in the sakai18n webservice, they need to be in lib for that to happen, not in a webapp (components or tool). I can just copy them into the grader's bundle, but then it's a duplication. Or I can just leave them as english as they were originally. Basically all the assignments grading strings would need to be synced with the grader's bundle.

File: assignment/tool/pom.xml [hunk 1]
```diff
@@ -36,10 +36,10 @@
             <groupId>${project.groupId}</groupId>
             <artifactId>sakai-assignment-api</artifactId>
         </dependency>
-        <dependency>
+        <!--dependency>
```
Comment (jesusmmp) [inline assignment/tool/pom.xml]: Please, remove this. Thanks!

**Comment by adrianfish** [comment]
Re transifex:
The options are: 1. start fresh with grader, adding translations as we go. 2. Copy the translations into grader from assignments and maintain as we go. 3. Make grader pick up the existing translations so they are shared between classic grading and new grader

File: assignment/tool/pom.xml (hunk 1 above)
Comment (adrianfish) [reply assignment/tool/pom.xml]: will do!

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java [hunk 2]
```diff
@@ -427,11 +427,18 @@ public AssignmentSubmission gradeSubmission(AssignmentSubmission submission, Str
...
             String siteId = (String) options.get("siteId");
 
+            Map<String, String> rubricsParams
+                = options.entrySet().stream().filter(e -> e.getKey().startsWith("rbcs"))
+                    .collect(Collectors.toMap(e -> e.getKey(), e -> (String) e.getValue()));
+
             // Persist the rubric evaluations
-            if (rubricsService.hasAssociatedRubric(RubricsConstants.RBCS_TOOL_ASSIGNMENT, submission.getAssignment().getId())){
+            if (!rubricsParams.isEmpty()) {
+            if (rubricsService.hasAssociatedRubric(RubricsConstants.RBCS_TOOL_ASSIGNMENT, submission.getAssignment().getId(), siteId)){
```
Comment (jesusmmp) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java]: Spacing...

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java [hunk 3]
```diff
@@ -427,11 +427,18 @@ public AssignmentSubmission gradeSubmission(AssignmentSubmission submission, Str
...
+                /*
```
Comment (jesusmmp) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java]: Please, remove this. Thanks!

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java (hunk 2 above)
Comment (adrianfish) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java]: good spot, forgot that.

**Comment by adrianfish** [comment]
Spanish, Basque and Catalonians! What do you think about the bundle? I was trying to make sure we picked up the existing bundle so we benefitted from previous i18n work and any future changes would appear in the new and current grading interfaces. I could just move them back but then we would have to make sure to copy the strings into the grader.properties bundle as we updated them in assignments.properties. Whatever makes sense, I am fine with.

File: webcomponents/tool/src/main/frontend/grader/sakai-grader.js [hunk 4]
```diff
@@ -386,43 +420,51 @@ class SakaiGrader extends gradableDataMixin(SakaiElement) {
...
-    editor.on("change", () => this.modified = true);
+    editor.on("change", () => this.modified = true );
     return editor;
   }
 
   toggleInlineFeedback(e, cancelling) {
 
-    if (!this.inlineFeedbackMode) {
-      this.inlineFeedbackMode = true;
+    //if (!this.inlineFeedbackMode) {
```
Comment (jesusmmp) [inline webcomponents/tool/src/main/frontend/grader/sakai-grader.js]: Please @adrianfish, could you remove 429, 431 and 438?. Thanks!

File: webcomponents/tool/src/main/frontend/grader/sakai-grader.js [hunk 5]
```diff
@@ -170,6 +191,9 @@ class SakaiGrader extends gradableDataMixin(SakaiElement) {
   renderSaved() {
     return html`<span class="saved fa fa-check-circle" title="${this.i18n["saved_successfully"]}" style="display: ${this.saved ? "inline" : "none"};"></span>`;
   }
+            //<a href="javascript:;" id="grader-rubric-link" @click=${this.toggleRubric} title="${this.assignmentsI18n["grading_rubric"]}">
```
Comment (jesusmmp) [inline webcomponents/tool/src/main/frontend/grader/sakai-grader.js]: Please, remove this one too. Thanks

**Comment by SedueRey** [comment]
> Spanish, Basque and Catalonians! What do you think about the bundle? I was trying to make sure we picked up the existing bundle so we benefitted from previous i18n work and any future changes would appear in the new and current grading interfaces. I could just move them back but then we would have to make sure to copy the strings into the grader.properties bundle as we updated them in assignments.properties. Whatever makes sense, I am fine with.

We just talked with @juanjmerono and he told us we could fix those new properties almost automatically so I approve this PR.

**Comment by jesusmmp** [comment]
LGTM!

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 7
           

Complexity increasing per file
==============================
- rubrics/tool/src/main/frontend/webcomponents/sakai-rubric-grading-button.js  5
         

Complexity decreasing per file
==============================
+ webcomponents/tool/src/main/frontend/grader/sakai-grader.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5124965&bid=16846229)

**Pull request merged**